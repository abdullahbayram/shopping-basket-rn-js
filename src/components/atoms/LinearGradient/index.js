import React from 'react';
import PropTypes from 'prop-types';
import { LinearGradient as Gradient } from 'expo-linear-gradient';
import globalStyles from '../../../globalStyles';

const LinearGradient = ({ colors, start = { x: 0, y: 0 }, end = { x: 1, y: 1 }, style = {} }) => (
  <Gradient colors={colors} start={start} end={end} style={[globalStyles.flex, style]} />
);

LinearGradient.propTypes = {
  colors: PropTypes.arrayOf(PropTypes.string).isRequired, // Gradient colors
  start: PropTypes.shape({ x: PropTypes.number, y: PropTypes.number }),
  end: PropTypes.shape({ x: PropTypes.number, y: PropTypes.number }),
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
};

export default LinearGradient;
