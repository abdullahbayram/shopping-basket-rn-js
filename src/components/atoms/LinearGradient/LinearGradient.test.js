import React from 'react';
import { render, screen } from '@testing-library/react-native';
import LinearGradient from '.';

describe('LinearGradient Component', () => {
  it('renders correctly with default props', () => {
    render(<LinearGradient colors={['#000000', '#ffffff']} />);
    const gradient = screen.getByTestId('linear-gradient');
    screen.getByTestId('linear-gradient');
    expect(gradient).toBeTruthy();
  });

  it('applies custom start and end props correctly', () => {
    const start = { x: 0.5, y: 0.5 };
    const end = { x: 1, y: 1 };
    render(<LinearGradient colors={['#123456', '#abcdef']} start={start} end={end} />);

    // The snapshot reveals that the LinearGradient component uses `startPoint` and `endPoint`
    // instead of `start` and `end`, and the values are represented as arrays ([x, y]) instead of objects ({x, y}).
    const gradient = screen.getByTestId('linear-gradient');
    expect(gradient.props.startPoint).toEqual([0.5, 0.5]);
    expect(gradient.props.endPoint).toEqual([1, 1]);
  });

  it('applies custom styles correctly', () => {
    const style = { margin: 10 };
    render(<LinearGradient colors={['#123456', '#abcdef']} style={style} />);

    const gradient = screen.getByTestId('linear-gradient');
    expect(gradient.props.style).toContainEqual(style);
  });

  // Snapshot Tests
  it('matches the snapshot with default props', () => {
    render(<LinearGradient colors={['#000000', '#ffffff']} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('matches the snapshot with custom start and end props', () => {
    const start = { x: 0.5, y: 0.5 };
    const end = { x: 1, y: 1 };
    const { toJSON } = render(<LinearGradient colors={['#123456', '#abcdef']} start={start} end={end} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('matches the snapshot with custom styles', () => {
    const style = { margin: 10 };
    render(<LinearGradient colors={['#123456', '#abcdef']} style={style} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
