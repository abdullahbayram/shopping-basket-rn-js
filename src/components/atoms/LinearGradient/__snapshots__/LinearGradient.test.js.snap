// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinearGradient Component matches the snapshot with custom start and end props 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  colors={
    [
      4279383126,
      4289449455,
    ]
  }
  endPoint={
    [
      1,
      1,
    ]
  }
  startPoint={
    [
      0.5,
      0.5,
    ]
  }
  style={
    [
      {
        "flex": 1,
      },
      {},
    ]
  }
  testID="linear-gradient"
/>
`;

exports[`LinearGradient Component matches the snapshot with custom styles 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  colors={
    [
      4279383126,
      4289449455,
    ]
  }
  endPoint={
    [
      1,
      1,
    ]
  }
  startPoint={
    [
      0,
      0,
    ]
  }
  style={
    [
      {
        "flex": 1,
      },
      {
        "margin": 10,
      },
    ]
  }
  testID="linear-gradient"
/>
`;

exports[`LinearGradient Component matches the snapshot with default props 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  colors={
    [
      4278190080,
      4294967295,
    ]
  }
  endPoint={
    [
      1,
      1,
    ]
  }
  startPoint={
    [
      0,
      0,
    ]
  }
  style={
    [
      {
        "flex": 1,
      },
      {},
    ]
  }
  testID="linear-gradient"
/>
`;
